sphinx_gallery/directives.py

```diff
@@ -77,3 +80,192 @@ def run(self):
+
+        return [image_node]
+
+
+def _parse_srcset(st):
+    """ parse st"""
+    entries = st.split(',')
+    srcset = {}
+    for entry in entries:
+        spl = entry.strip().split(' ')
+        if len(spl) == 1:
+            srcset[0] = spl[0]
+        elif len(spl) == 2:
+            mult = spl[1][:-1]
+            srcset[float(mult)] = spl[0]
+    return srcset
+
+
+def visit_imgsrcset_html(self, node):
+
+    if node['srcset'] is None:
+        self.visit_image(node)
+        return
+
+    imagedir, srcset = _copy_images(self, node)
+
+    # /doc/examples/subd/plot_1.rst
+    docsource = Path(self.document['source'])
+    # /doc/
+    srctop = self.builder.srcdir
+    # examples/subd/plot_1.rst
+    relsource = docsource.relative_to(srctop)
+
+    # /doc/build/html
+    desttop = Path(self.builder.outdir)
+    # /doc/build/html/examples/subd
+    dest = (desttop / relsource).parent
+
+    # ../../_images
+    imagerel = Path(os.path.relpath(imagedir, dest))
+
+    # make srcset str.  Need to change all the prefixes!
+    srcsetst = ''
+    for mult in srcset:
+        nm = Path(srcset[mult][1:]).name
+        # ../../_images/plot_1_2_0x.png
+        relpath = Path(imagerel, nm)
+        srcsetst += f'{relpath}'
+        if mult == 0:
+            srcsetst += ', '
+        else:
+            srcsetst += f' {mult:1.1f}x, '
+    # trim trailing comma and space...
+    srcsetst = srcsetst[:-2]
+
+    # make uri also be relative...
+    uri = Path(imagerel, Path(node['uri'][1:]).name)
+
+    alt = node['alt']
+    if node['class'] is not None:
+        classst = node['class'][0]
+        classst = f'class = "{classst}"'
+    else:
+        classst = ''
+
+    html_block = (f'<img src="{uri}" srcset="{srcsetst}" alt="{alt}"' +
+                  f' {classst}/>')
+    self.body.append(html_block)
+
+
+def visit_imgsrcset_latex(self, node):
+
+    if node['srcset'] is not None:
+        imagedir, srcset = _copy_images(self, node)
+        maxmult = -1
+        # choose the highest res version for latex:
+        for key in srcset.keys():
+            maxmult = max(maxmult, key)
+        node['uri'] = str(Path(srcset[maxmult]).name)
+
+    self.visit_image(node)
+
+
+def _copy_images(self, node):
+    srcset = _parse_srcset(node['srcset'])
+
+    # where the sources are.  i.e. myproj/source
+    srctop = self.builder.srcdir
+
+    # copy image from source to imagedir.  This is
+    # *probably* supposed to be done by a builder but...
+    # ie myproj/build/html/_images
+    imagedir = Path(self.builder.outdir, self.builder.imagedir)
+    os.makedirs(imagedir, exist_ok=True)
+
+    # copy all the sources to the imagedir:
+    for mult in srcset:
+        abspath = Path(srctop, srcset[mult][1:])
+        shutil.copyfile(abspath, imagedir / abspath.name)
+
+    return imagedir, srcset
+
+
+def depart_imgsrcset_html(self, node):
+    pass
+
+
+def visit_srcset_other(self, node):
+    if node['uri'][0] == '/':
```

Comment:
and this will only work if all `uri`s are Posix flavored (which they should / probably need to be in principle I think)